### ISSUE TITLE
Example configuration

### ISSUE THREAD

**Issue by colelogan** [ShaMan123/react-native-sketch-canvas#14]
I’ve been trying to run your code on my emulator for a while. 

I’m trying to run the CanvasExample synced canvas. I tried following your directions but would not run on the emulator. Then I copied some files over to a ngx-create-react-native-app project and I still can’t figure it out.

I’m familiar with react but I just don’t know how to get all the configs done right. I really like your code too. I would really appreciate help if any.

**Comment by ShaMan123** [comment]
What errors are you getting?
What are the values of the file `example.config.json`?
What steps did you do?

**Comment by colelogan** [comment]
the values of  ``` example.config.json ``` are. 
``` 

{
	"useBaseImplementation": false,
	"useLocalReanimatedModule": true
}

```

The same version of reanimated pulled from your github and RN-reanimated-canvas are in the same folder:

![Root_Projects_Folder_with-react-native-reanimated](https://user-images.githubusercontent.com/20275733/106376952-60f1f280-634e-11eb-94c0-f48f87c8b170.PNG)

Here is the error code I'm getting:

![Error_located_in_StackViewLayout](https://user-images.githubusercontent.com/20275733/106377002-b62e0400-634e-11eb-8384-dd74010aaa9f.PNG)

I don't know what do about that at the moment.

**Comment by ShaMan123** [comment]
I want to see your `CanvasExample/package.json`

**Comment by ShaMan123** [comment]
Ohh and try to run `npm run android -- --reset-cache` from the example folder

**Comment by colelogan** [comment]
sure

```
{
  "name": "CanvasExample",
  "version": "0.0.1",
  "private": true,
  "main": "index.ts",
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "postinstall": "node ../scripts/touchExample.js",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@react-navigation/web": "^1.0.0-alpha.9",
    "lodash": "^4.17.15",
    "react": "16.9.0",
    "react-native": "0.61.5",
    "react-native-gesture-handler": "github:ShaMan123/react-native-gesture-handler#drag-gesture-handler",
    "react-native-reanimated": "github:ShaMan123/react-native-reanimated",
    "react-native-screens": "^2.3.0",
    "react-native-view-shot": "^3.0.2",
    "react-navigation": "^4.0.10",
    "react-navigation-stack": "^1.10.3",
    "typescript": "^3.7.2"
  },
  "devDependencies": {
    "@babel/core": "^7.6.2",
    "@babel/plugin-transform-modules-commonjs": "^7.7.0",
    "@babel/runtime": "^7.6.2",
    "@react-native-community/eslint-config": "^0.0.5",
    "@types/jest": "^24.0.22",
    "@types/react": "^16.9.11",
    "@types/react-native": "^0.60.22",
    "@types/react-test-renderer": "^16.9.1",
    "babel-jest": "^24.9.0",
    "babel-plugin-module-resolver": "^3.2.0",
    "eslint": "^6.5.1",
    "jest": "^24.9.0",
    "metro-react-native-babel-preset": "^0.56.0",
    "react-test-renderer": "16.9.0"
  },
  "jest": {
    "preset": "react-native"
  }
}
```

**Comment by colelogan** [comment]
ok I'll try that

**Comment by ShaMan123** [comment]
If that doesn't work try updating `yarn add react-native-gesture-handler`. The error seems to surface there but not sure if it's the origin.

**Comment by colelogan** [comment]
It's working hurrayy!

![is-working](https://user-images.githubusercontent.com/20275733/106378056-49b70300-6356-11eb-9915-f0159deb0fb4.PNG)

**Comment by colelogan** [comment]
I have one more question. How would you configure base path? I tried playing with it for a couple minutes setting localpath to false and base path to true. Is there something I'm missing?

![basepath-config-fail](https://user-images.githubusercontent.com/20275733/106378163-36f0fe00-6357-11eb-900e-6946f6432b19.PNG)

**Comment by ShaMan123** [comment]
Great!
Important notice:
There a 3 approaches to handle touch in this repo:
1. `JS` - bad performance (`terrylinla`'s approach)
2. `reanimated` - works only on android (my PR), `reanimated` are moving to a different approach in their `v2` so it seems a waste to support iOS...
3. `native` - needs to be implemented (android requires a little tweak to get it working). However, the last time I tested it, it didn't play nicely with heavy customization **but** a fix I pushed to RNGH might make the difference.

**Comment by ShaMan123** [comment]
> I have one more question. How would you configure base path? I tried playing with it for a couple minutes setting localpath to false and base path to true. Is there something I'm missing?
> 
Yes. You must re-build the app `npm run android -- --reset-cache` for it to make an affect.

**Comment by ShaMan123** [comment]
The base impl tests the repo without `reanimated` or `RNGH` as deps

**Comment by ShaMan123** [comment]
closing

**Comment by colelogan** [comment]
Oh those are really interesting points. And yes your project is mostly working but there are some errors on two screens synced canvas and the updating screen. I would have to check your RNGH push out. I thought base config was running animated from node_modules but I understand now. 

It's seems like you really know what you are doing with react & react-native like alot... thats very cool. 

Are you a contractor or do you work for a company?

**Comment by ShaMan123** [comment]
Thanks.
It was hard and frustrating getting this to work.
I guess you could say I'm a contractor.
Could you share the errors?

**Comment by colelogan** [comment]
Yes let me send you the screenshots

**Comment by colelogan** [comment]
This error is from the synced canvas screen:

![sync-two-canvas](https://user-images.githubusercontent.com/20275733/106379003-c3062400-635d-11eb-9c2c-0b26b84058d1.PNG)

This error is from the imperative updating screen:

![imperative-updating-error](https://user-images.githubusercontent.com/20275733/106379006-c9949b80-635d-11eb-9148-76480b2bd8c6.PNG)

**Comment by colelogan** [comment]
Do you have any apps published on the app store?

**Comment by colelogan** [comment]
The Imperative Updating screens canvas draw worked for a little bit before it errored me.

The Synced Canvases Screen error is weird. I think I might need to update the Gradle version.

**Comment by ShaMan123** [comment]
both of the errors seem to be caused by a path that is undefined

**Comment by ShaMan123** [comment]
> Do you have any apps published on the app store?

I worked on an app called [Autodidact](https://play.google.com/store/apps/details?id=io.autodidact) but abandoned it after a major pivot. It is open for testers and is quite basic but uses this repo.

**Comment by ShaMan123** [comment]
> both of the errors seem to be caused by a path that is undefined

Some logic error I believe - perhaps in the example, but not sure about it

**Comment by ShaMan123** [comment]
What are you working on?

**Comment by colelogan** [comment]
I'm working on a project where you and another person draw together over the internet. Like FaceTime and draw.

**Comment by colelogan** [comment]
What do you think of the idea?

**Comment by ShaMan123** [comment]
The major design consideration would be how live you need/want it to be.
2 options depending on your needs:
1. Live to the touch - meaning that you need to stream the touches as they occur and display them immediately.
2. Live to the path - meaning that you will stream an entire path after the touch has ended and perhaps display it to the other user with some kind of animation. 

Viability:
1. Possible, will require more work on native (I would create a component that handles everything in native including the database updates, this way you won't need to cross the js bridge at all) or alternatively **reanimated V2** might be a feasible option for drawing imperatively on both platforms. I was testing V2 a while back in order to migrate this repo to V2, it was still very new.... I don't know it's state nowadays. If it's in good shape and shows good results migrating should be easy.
2. Sounds easier to implement but may be harmful to the UX/UI (may seem laggy) and to the main purpose of the app. In order to create an animation you could use reanimated V2 (or my fork or common js - depending on performance) or you could add a small native method or prop that handles it (will be better for perf and more stable in case reanimated changes dramatically).

**Comment by ShaMan123** [comment]
combining the 2 approaches sounds powerful

**Comment by ShaMan123** [comment]
Anyways make sure to run stress tests before you move on or make a decision because this is a flaky piece of UI.

**Comment by colelogan** [comment]
Makes sense. Ok, yes I would want to stream to touch. I think the v2 with the prop handling would be better for this use case. I'm terrified of the thought of having to manage native code on both platforms. 

I will look into v2 more and look into some native code for this.

**Comment by colelogan** [comment]
Synced Canvas quick fix...

```pathIDMem``` isn't an export anymore in ```RCanvasModule``` due to a change in the git version.

![pathidmem-does-not-exist-in-RCanvasModule](https://user-images.githubusercontent.com/20275733/106539629-31a8c600-64b3-11eb-9eb2-4fa258f26154.PNG)

what I did to fix this issue is change the ```id``` variable a new Value

![change-id-value](https://user-images.githubusercontent.com/20275733/106539635-340b2000-64b3-11eb-86a4-32cbd4f0fa6e.PNG)

and changed the logic in the condition block, ```createPathID()``` doesn't work like it should so I did  ```set(id, add(id, 1)),```

![what-i-did](https://user-images.githubusercontent.com/20275733/106540716-556d0b80-64b5-11eb-8de7-29a8a7240080.PNG)

and it works

![it-should-work-like-this](https://user-images.githubusercontent.com/20275733/106539643-38373d80-64b3-11eb-8ca7-80cc7a8e853d.PNG)

**Comment by colelogan** [comment]
Just to confirm sha, you said this build cannot be supported in iOS?

**Comment by ShaMan123** [comment]
Good job!
It isn't supported in iOS but eith work it can be.

בתאריך יום ג׳, 2 בפבר׳ 2021, 06:07, מאת Cole L. ‏<notifications@github.com>:

**Comment by colelogan** [comment]
And thats with native support like written in Xcode for the iOS app to handle react right? It's like a Xcode module handler right?

**Comment by colelogan** [comment]
I just looked into the iOS folder and saw the code that seems like a ton of work. What else must be done?

**Comment by ShaMan123** [comment]
You are right. For someobe who is fluent in iOS it ahouldn't take more than
a day's work

בתאריך יום ג׳, 2 בפבר׳ 2021, 11:29, מאת Cole L. ‏<notifications@github.com>:

**Comment by colelogan** [comment]
Ok I appreciate the feedback. I'm going to remove the unnecessary comments on this thread for easy readability.

**Comment by colelogan** [comment]
And I will reach out to you in the future because you're good and you give good feedback on certain points. Anyways I appreciate the help I really do

**Comment by ShaMan123** [comment]
Cheers.
You're welcome to PR the fix you made to the example.

בתאריך יום ג׳, 2 בפבר׳ 2021, 20:17, מאת Cole L. ‏<notifications@github.com>:

**Comment by colelogan** [comment]
I apologize to keep dragging this conversation on and I know I told you what my idea kinda is, I told you a half truth. I tried looking into the code myself and figured that it's out of my scope. 

I know you might be busy with other things right now but, Hypothetically Sha, if I received funding, How much would you cost?

**Comment by colelogan** [comment]
I can get funding for 20 - 30,000 USD for the FaceTime with canvas draw functionality. We can talk more later. Would that be something that you can do?

**Comment by ShaMan123** [comment]
Let's move to email:
shacharnen@gmail.com
